go/proto/metald/v1/vm.proto

```diff
@@ -0,0 +1,232 @@
+syntax = "proto3";
+
+package metald.v1;
+
+option go_package = "github.com/unkeyed/unkey/go/gen/proto/metald/v1;metaldv1";
+
+import "google/protobuf/timestamp.proto";
+import "metald/v1/network.proto";
+import "metald/v1/storage.proto";
+
+// VM lifecycle states
+enum VmState {
+  VM_STATE_UNSPECIFIED = 0;
+  VM_STATE_CREATED = 1;
+  VM_STATE_RUNNING = 2;
+  VM_STATE_PAUSED = 3;
+  VM_STATE_SHUTDOWN = 4;
+}
+
+// Unified VM configuration that works across different hypervisors
+message VmConfig {
+  // CPU configuration
+  int32 cpu = 1;
+
+  // Memory configuration
+  int64 memory = 2;
+
+  // Boot configuration
+  string boot = 3;
+
+  // Network configuration
+  string network_config = 4;
+
+  // Console configuration
+  ConsoleConfig console = 5;
+
+  // Metadata and labels
+  map<string, string> metadata = 6;
+
+  // VM Identifier
+  string id = 7;
+}
+
+message ListVmsRequest {
+  // Optional filter by state
+  repeated VmState state_filter = 1;
+
+  // Pagination
+  int32 page_size = 2;
+  string page_token = 3;
+}
+
+message ListVmsResponse {
+  repeated VmInfo vms = 1;
+  string next_page_token = 2;
+  int64 total_count = 3;
+}
+
+// Request/Response messages
+message CreateVmRequest {
+  // Generated unique identifier for the VM
+  string vm_id = 1;
```

Comment:
@imeyer I'm very confused, you asked me to have the vm id be created inside metald and then 2 days later you create this schema where it's passed in?

which one is it now?